### PR TITLE
Increase number of forward tokens for isolation tasklists

### PR DESCRIPTION
**Pull request by Shaddoll**
**What changed?**
Increase number of forward tokens for isolation tasklists

**Why?**
Fix the latency degradation for scalable tasklists with isolation

**How did you test it?**
staging2 test

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/matching/forwarder.go [hunk 1]
```diff
@@ -91,19 +93,21 @@ func newForwarder(
 	taskListID *taskListID,
 	kind types.TaskListKind,
 	client matching.Client,
+	isolationGroups []string,
```
Comment (davidporter-id-au) [inline service/matching/forwarder.go:96]: Is this intended to be that private-package type?

File: service/matching/forwarder.go [hunk 2]
```diff
@@ -91,19 +93,21 @@ func newForwarder(
...
 ) *Forwarder {
 	rpsFunc := func() float64 { return float64(cfg.ForwarderMaxRatePerSecond()) }
 	fwdr := &Forwarder{
 		cfg:                   cfg,
 		client:                client,
 		taskListID:            taskListID,
 		taskListKind:          kind,
-		outstandingTasksLimit: int32(cfg.ForwarderMaxOutstandingTasks()),
-		outstandingPollsLimit: int32(cfg.ForwarderMaxOutstandingPolls()),
+		outstandingTasksLimit: int32(cfg.ForwarderMaxOutstandingTasks() * (len(isolationGroups) + 1)),
```
Comment (davidporter-id-au) [inline service/matching/forwarder.go:104]: Mind adding a comment here as to the intuition of this?

File: service/matching/forwarder.go [hunk 3]
```diff
@@ -57,6 +57,8 @@ type (
 		// todo: implement a rate limiter that automatically
 		// adjusts rate based on ServiceBusy errors from API calls
 		limiter *quotas.DynamicRateLimiter
+
+		isolationGroups []string
```
[the remark this answers is not in the document]
Comment (Shaddoll) [reply service/matching/forwarder.go:61]: it is used

File: service/matching/forwarder.go (hunk 1 above)
Comment (Shaddoll) [reply service/matching/forwarder.go:96]: it's allisolationgroups if isolation is enabled, it's nil if not

File: service/matching/forwarder.go (hunk 2 above)
Comment (Shaddoll) [reply service/matching/forwarder.go:104]: there is an intuition in line 264

File: service/matching/forwarder.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (davidporter-id-au) [reply service/matching/forwarder.go:61]: sorry, I was tired... this is not a type declaration like I thought it was, it's a field on a struct 🤦

**Review by davidporter-id-au — APPROVED** [review]

**Review by davidporter-id-au — APPROVED** [review]

**Pull request merged**